### ISSUE TITLE
Streamsaver does not clean up iframes

### ISSUE THREAD

**Issue by acuntex** [jimmywarting/StreamSaver.js#134]
When using StreamSaver.js in a long living SPA, you see that StreamSaver.js does not cleanup the iframes, thus poluting the DOM.

Repro: Just download and look at the dom :-)

**Comment by jimmywarting** [comment]
I tried to clean it up before but for some reason it didn't work so well, 
Don't remember what it was exactly, in some edge cases the download never started if the iframe where removed before any bytes have been written to the iframe - so it would just be safer to leave it there, all it is is an empty iframe - not consuming any dom, js or css 

backlog: https://github.com/jimmywarting/StreamSaver.js/issues/94

**Comment by acuntex** [comment]
Just an idea: Can we somehow tag the iframes with a custom attribute and a timestamp? So that an application can identify these iframes and clean up "old" iframes.

I've noticed performance issues if there are more than 100 iframes. Not 100% sure if it's because of this but this is my best guess.

**Comment by jimmywarting** [comment]
sure, we could "tag" them with something. `<iframe data-streamsaver>` maybe?
otherwise another solution is to get them with src css-selector

`$(`iframe[src^="${streamSaver.mitm.replace(/mitm.*/, location.hostname)}"]`)`

^ means startsWith

**Comment by acuntex** [comment]
Yes, but i would maybe also include a timestamp... just in case so that an automatic cleanup by the application does not kill the current download.

**Comment by jimmywarting** [comment]
fyi when native file system gets implemented, then we might not need iframes or service workers :)

**Comment by jimmywarting** [comment]
> more than 100 iframes

why don't you zip everything?

**Comment by acuntex** [comment]
> fyi when native file system gets implemented, then we might not need iframes or service workers :)
yay :-)

> why don't you zip everything?
It's a timing issue. This would be only possible if all files would be present at the same time. In a long running application this might not be the case :-(
